src/java/org/apache/cassandra/index/sai/iterators/InMemoryPartitionIterator.java

```diff
@@ -16,12 +16,11 @@
  * limitations under the License.
  */
 
-package org.apache.cassandra.index.sai.utils;
+package org.apache.cassandra.index.sai.iterators;
```

Comment:
In ASF this one and `InMemoryUnfilteredPartitionIterator` are in `utils, but I guess it makes sense to have them in `iterators`, +1.

Reply:
For some reason I was blind and haven't noticed the files in ASF's `utils`. As one of the goal of this PR is to make it easier to rebase from 5.0, then it might make sense to revert.
There is no good consistency for locations of iterators. `memtable` folder contains iterators related to memtables.
Let me to know if you will change your opinion to revert the change and move the files back to `utils`.

Reply:
I don't have a clear opinion on this one. It might make sense to put all the iterators together, but it's also nice to avoid merge conflicts. Also, one could argue that these two iterators are only used by `org.apache.cassandra.index.sai.plan.VectorTopKProcessor`, so they might as well be in the `org.apache.cassandra.index.sai.plan` package. I guess that in doubt I'll go back to the ASF location, but I'll be fine with whatever you prefer.

Reply:
I moved them back to `utils` as I tried to focus this PR on replicating ASF structure. Thought `utils` is not natural place for them, but this is likely a separate discussion and I am not ready to cover in the scope of this PR.